src/benchmarks/gc/GC.Infrastructure/GC.Analysis.API/CPUAnalysis/CPUAnalysisExt.cs

```diff
@@ -246,10 +246,14 @@ public static List<CPUInfo> GetPerGCMethodCost(this CPUProcessData processData,
 
         public static void SetSourcePath(this CPUProcessData processData, string sourcePath)
             => processData.StackView.SymbolReader.SourcePath = sourcePath;
-        public static string Annotate(this CPUProcessData processData, string methodName)
+        public static string? Annotate(this CPUProcessData processData, string methodName)
         {
-            SourceLocation location = processData.StackView.GetSourceLocation(methodName, out var @out);
-            SortedDictionary<int, float> metricsOnLine = @out;
+            if (!processData.StackView.TryGetSourceLocation(methodName, out SourceLocation? location, out SortedDictionary<int, float>? metricsOnLine))
+            {
+                // TODO: Log an error.
```

Comment:
Would throwing an exception make more sense?

Reply:
I copied this from `GetSourceLocation`.  I'm happy to add a throw, but I wasn't sure what the model was here (which exception, etc.).  Any thoughts on that?

(also this is just to remove one of 100s of null warnings, so if this is complicated, I should just revert and leave it for another time)

Reply:
I see - yes, I filched this from the PerfView source. I think it's fine, as you have it. We can revisit later.